lib/p2p/Peer.ts

```diff
@@ -265,16 +270,45 @@ class Peer extends EventEmitter {
         resolve();
       };
 
-      const onTimeout = () => {
+      const onError = (err: Error) => {
         cleanup();
-        reject(new Error('Connection timed out.'));
-      };
 
-      this.socket!.once('connect', onConnect);
+        if (!retry) {
+          this.close();
+          reject(err);
+          return;
+        }
+
+        retryDelay = !retryDelay
+          ? Peer.CONNECTION_RETRIES_MIN_DELAY
+          : Math.min(Peer.CONNECTION_RETIES_MAX_DELAY, retryDelay * 2);
+
+        if (Date.now() - startTime + retryDelay > Peer.CONNECTION_RETIES_MAX_PERIOD) {
+          this.close();
+          reject(new Error(`${retries} connection retries failed`));
+          this.emit('ban');
+          return;
+        }
+
+        this.logger.info(
```

Comment:
I think `debug` or perhaps `verbose` would be better here, this feels like we'd have a lot of messages in the logs that aren't particularly useful.